jumpscale/core/base/store/etcd.py

```diff
@@ -0,0 +1,108 @@
+import etcd3
+import json
+
+from . import ConfigNotFound, EncryptedConfigStore, EncryptionMode
+
+from .serializers import JsonSerializer
+
+
+class EtcdStore(EncryptedConfigStore):
+    """
+    EtcdStore store is an EncryptedConfigStore
+
+    It saves the data in etcd and configuration for etcd comes from `config_env.get_store_config("etcd")`
+    """
+
+    def __init__(self, location):
+        """
+        create a new etcd store, the location given will be used to generate keys
+
+        this keys will be combined to get/set instance config
+
+        Args:
+            location (Location)
+        """
+        super().__init__(location, JsonSerializer())
+        etcd_config = self.config_env.get_store_config("etcd")
+        hostname = etcd_config.get("hostname", "localhost")
+        port = etcd_config.get("port", 2379)
+        self.etcd_client = etcd3.client(host=hostname, port=port)
+
+    def get_key(self, instance_name):
+        """
+        get a key for an instance
+
+        this will return a dot-separated key derived from current location
+
+        Args:
+            instance_name (str): name
+
+        Returns:
+            str: key
+        """
+        return ".".join([self.location.name, instance_name])
+
+    def read(self, instance_name):
+        """
+        read instance config from etcd
+
+        Args:
+            instance_name (name): name
+
+        Returns:
+            str: return json object
+        """
+        key = self.get_key(instance_name)
+        if not self.etcd_client.get(key)[0]:
```

Comment:
won't this raise if .get(key) is empty?

Reply:
no, if the key is not found it will still return a tuple with (None,None) and so the first item in the tuple is the one that is empty not the whole object